backend/multi/backend.c

```diff
@@ -28,13 +30,22 @@ static bool multi_backend_start(struct wlr_backend *wlr_backend) {
 	return true;
 }
 
+static void subbackend_state_destroy(struct subbackend_state *sub) {
+	wl_list_remove(&sub->input_add.link);
+	wl_list_remove(&sub->input_remove.link);
+	wl_list_remove(&sub->output_add.link);
+	wl_list_remove(&sub->output_remove.link);
+	wl_list_remove(&sub->backend_destroy.link);
+	wl_list_remove(&sub->link);
+	free(sub);
+}
+
 static void multi_backend_destroy(struct wlr_backend *wlr_backend) {
 	struct wlr_multi_backend *backend = (struct wlr_multi_backend *)wlr_backend;
-	wl_list_remove(&backend->display_destroy.link);
 	struct subbackend_state *sub, *next;
 	wl_list_for_each_safe(sub, next, &backend->backends, link) {
+		// XXX do we really want to take ownership over added backends?
```

Comment:
Yes

Reply:
Ok, well the implication is that backends cannot belong to multiple multibackends which needs to be made clear to people. Dunno why anyone would want more than one multibackend though.